internal/services/containerapps/helpers/container_apps.go

```diff
@@ -2257,21 +2288,36 @@ func SecretsDataSourceSchema() *pluginsdk.Schema {
 	}
 }
 
-func ExpandContainerSecrets(input []Secret) *[]containerapps.Secret {
+func ValidateContainerSecret(s Secret) error {
+	if s.KeyVaultUrl != "" && s.Value != "" {
+		return fmt.Errorf("key vault url and value are mutually exclusive")
+	}
+	if s.KeyVaultUrl != "" && s.Identity == "" {
+		return fmt.Errorf("must supply identity for key vault url")
+	}
```

Comment:
Shall we also check for `s.KeyVaultUrl == "" && s.Identity != ""`?

Reply:
added